### PR TITLE
[Filebeat] Add CustomString mapping to CEF for Forcepoint NGFW

### LINKED ISSUE #14663: [Filebeat] Add module for Forcepoint firewall logs
Add a fileset for ingesting logs from Forcepoint Next Generation Firewall (NGFW). The firewall can export CEF data over syslog (among other formats). So the fileset should be able to leverage the [syslog input](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-syslog.html) and the [`decode_cef` processor](https://www.elastic.co/guide/en/beats/filebeat/current/processor-decode-cef.html) to ingest the logs and then in a module apply the appropriate transformations to match ECS as much as possible.

- [Forcepoint: How to forward SMC log and audit data to external syslog or SIEM servers ](https://support.forcepoint.com/KBArticle?id=000015002)

### PR DESCRIPTION
**Pull request by leehinman**
- map CustomString1 to rule.id when CustomString1Label == RuleID

Closes #14663

### CODE REVIEW

**Comment by leehinman** [comment]
> Seems like a nice simple enhancement to the module to add support for Forcepoint logs. Are there any more custom fields to add?

Not in the examples I have.  I have a request out to see if there are any more.

> 
> We'll want to document that the CEF module works for Forcepoint NGFW. I propose adding a section to the existing cef module documentation that shows it works for forcepoint NGFW and was tested against logs from version X and specifying that it needs to be configured to send CEF logs to the module's input address, etc.

will do

File: x-pack/filebeat/module/cef/log/config/input.yml [hunk 1]
```diff
@@ -24,3 +24,14 @@ processors:
   - decode_cef:
       field: event.original
   - community_id:
+  - script:
```
[the remark this answers is not in the document]
Comment (leehinman) [reply x-pack/filebeat/module/cef/log/config/input.yml]: Good idea.  I was waffling on this, the ruleid match seemed pretty generic, but the when is safer.

[the remark this answers is not in the document]
Comment (andrewkroh) [reply x-pack/filebeat/module/cef/log/config/input.yml]: On a related note, I think having support in the processor for directly creating key-value pairs from the custom label and value fields would be a nice future enhancement. Like automatically create `cef.custom.string.mylabel: "myvalue"` and if the label is an ECS field like `rule.id` (and the value is the proper data type) then just go ahead and populate the field.

File: x-pack/filebeat/module/cef/log/config/input.yml [hunk 2]
```diff
@@ -24,3 +24,22 @@ processors:
...
+      lang: javascript
+      source: >-
+        function forcepoint_mappings(evt) {
```
Comment (andrewkroh) [inline x-pack/filebeat/module/cef/log/config/input.yml]: I liked what you had before, but just wanted a `when` condition added. The primary benefit to `when` is that it avoids invoking the javascript VM at all which I _assume_ is a performance improvement over doing the check in JS for non-forcepoint logs.

**Pull request merged**